CMakeLists.txt

```diff
@@ -53,7 +53,7 @@ set(RAJA_CXX_STANDARD_FLAG "default" CACHE STRING "Specific c++ standard flag to
 option(ENABLE_TBB "Build TBB support" Off)
 option(ENABLE_TARGET_OPENMP "Build OpenMP on target device support" Off)
 option(ENABLE_CLANG_CUDA "Use Clang's native CUDA support" Off)
-option(ENABLE_EXTERNAL_CUB "Use an external cub for scans" Off)
+option(ENABLE_EXTERNAL_CUB "Use an external cub for scans" On)
```

Comment:
This should only be on if CUDA is enabled or HIP building for nvidia hardware, correct?

Reply:
This is always used in places guarded by ENABLE_CUDA.
This PR does not include support for HIP targeting CUDA.

Reply:
OK. It seems a little froggy to me though. Does it seem confusing to you?

Reply:
I know what you mean, I can put this option in a conditional if that makes you feel better.

Reply:
As in, if ENABLE_CUDA option On else option Off

Reply:
@davidbeckingsale @trws @ajkunen do you have any opinions on this?

Reply:
Is this still doing anything? We had it to override the vendored version, but are no longer using a vendored version.

Reply:
Currently if its turned off we install cub from the submodule. We remove the cub submodule entirely and only support external cub.

Reply:
It seems we want to support three alternatives: 1) use cub submodule, 2) use external cub that user provides, and 3) use cub associated with the given nvcc toolchain. Correct, or do we just want 2 and 3?

To me, it would be clearer if option names and CMake logic follows the options we support. Here, "EXTERNAL_CUB" only captures one option and using it for other choices seems confusing.